Src/main.c

```diff
@@ -102,19 +102,30 @@ int main(void)
 
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
+
+  HAL_TIM_PWM_START(&htim16, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+    // Read value from potentiometer
+    HAL_ADC_Start(&hadc1);
+    HAL_ADC_PollForConversion(&hadc1, HAL_MAX_DELAY);
+    potentiometer_value = HAL_ADC_GetValue(&hadc1);
+
+    servo_pulse = ((potentiometer_value / 4095) * (2-1) + 1) / 0.2 * 60000; //Scale the potentiometer value to the right pulse value to send to the motor
```

Comment:
I would put a float cast here just to be safe - e.g ((float)(potentiometer_value)/4095.0)

Also, keep everything in terms of ms (20 not 0.2) (2 and 1 are fine)

Also, I believe you should use variables for these values (high, low, period) to the code more robust later down. That way in practice you could change the high, low, period easier.